src/local.jl

```diff
@@ -32,25 +32,41 @@ function localzone()
 
         # Try getting the time zone from the "TZ" environment variable
         # http://linux.die.net/man/3/tzset
+        #
+        # Note: The macOS "man tzset 3" states some additional information about the colon
+        # being optional:
+        #
+        # > If TZ appears in the environment and its value begins with a colon (`:'), the
+        # > rest of its value is used as a pathname of a tzfile(5)-format file from which to
+        # > read the time conversion information. If the first character of the pathname is
+        # > a slash (`/'), it is used as an absolute pathname; otherwise, it is used as a
+        # > pathname relative to the system time conversion information directory.
+        # >
+        # > If its value does not begin with a colon, it is first used as the pathname of a
+        # > file (as described above) from which to read the time conversion information. If
+        # > that file cannot be read, the value is then interpreted as a direct
+        # > specification (the format is described below) of the time conversion
+        # > information.
         if haskey(ENV, "TZ")
             name = ENV["TZ"]
 
-            # When the TZ format starts with a colon this indicates that the time zone information
-            # should be read from a file.
+            # If the TZ format starts with a colon we'll prefer using the time zone
+            # information from the specified file.
             if startswith(name, ':')
-                name = name[2:end]
+                name = name[2:end]  # Name is either an relative or absolute path
             else
-                return parse_tz_format(name)
+                tz = tryparse_tz_format(name)
```

Comment:
This logic is reversed from the manual-specified logic, but I'm fine with that if you are

Reply:
The Linux tzset man page lists the filespec format as the the "second" format (http://man7.org/linux/man-pages/man3/tzset.3.html). There doesn't seem to be a canonical specification so I'll keep this as they are.